digAddressExtractor/address_extractor.py

```diff
@@ -0,0 +1,204 @@
+
+
+def processFile(input_file_name,keywords,finalAddressData):
+    input_data=""
+    with open(input_file_name) as data_file:
+        input_data=json.load(data_file)
+    extracted_address=[]
+    for each_type in input_data:
+        for each_a in input_data[each_type]:
+            temp={}
+            temp["input"]=each_a
+            addresses=Set()
+            for each_keyword in keywords:
+                p = re.compile(r'\b%s\b' % each_keyword.lower(), re.I)
+                m=p.search(each_a.lower())
+                if m!=None:
+                    addresses=extractAddress(each_a,p,each_keyword,addresses)
+            temp["address"]=list(addresses)
+            finalAddressData.append(temp)
+
+"""
+Input: Text String
+Output: Json object containing input text string with list of associated present addresses
+
+Uses default keywords embedded in script file
+"""
+def getAddressFromString(text_string):
+    # temp={}
+    #temp["input"]=text_string
+    addresses=Set()
+    for each_keyword in keywords:
+        p = re.compile(r'\b%s\b' % each_keyword.lower(), re.I)
+        m=p.search(text_string.lower())
+        if m!=None:
+            extractAddress(text_string,p,each_keyword,addresses)
+    #temp["address"]=list(addresses)
+    return addresses
+
+"""
+Input: Text String and keyword python list ex: ["ave","street"] etc.
+Output: Json object containing input text string with list of associated present addresses
+
+Uses keywords list passed as an parameter
+"""
+def getAddressFromStringType(text_string,keywords):
+    temp={}
+    temp["input"]=text_string
+    addresses=Set()
+    for each_keyword in keywords:
+        p = re.compile(r'\b%s\b' % each_keyword.lower(), re.I)
+        m=p.search(text_string.lower())
+        if m!=None:
+            extractAddress(text_string,p,each_keyword,addresses)
+    temp["address"]=list(addresses)
+    return temp
+
+"""
+Input: File name and keyword python list ex: ["ave","street"] etc.
+Output: Json object containing list of all input text strings with list of associated present addresses
+"""
+def getAddressFromFile(file_name,keywords):
+    finalAddressData=[]
+    processFile(file_name,keywords,finalAddressData)
+    return finalAddressData
+
+
+
+import copy 
+import types
+from digExtractor.extractor import Extractor
+
+class AddressExtractor(Extractor):
+
+    def __init__(self):
+        self.renamed_input_fields = ['text']  # ? renamed_input_fields
+
+    def extract(self, doc):
+        return getAddressFromStringType(doc['text'], keywords)
+        
+    def get_metadata(self):
+        return copy.copy(self.metadata)
+
+    def set_metadata(self, metadata):
+        self.metadata = metadata
+        return self
+
+    def get_renamed_input_fields(self):
+        return self.renamed_input_fields
+
+    def set_renamed_input_fields(self, renamed_input_fields):
```

Comment:
There's no need to copy this method since it's implemented in the parent Extractor.